src/clj/game/core/costs.clj

```diff
@@ -98,21 +101,36 @@
                    (-> (card-def %) :interactions :pay-credits ((fn [x] (:custom-amount x 0))))))
           (reduce +))))
 
+(defn- eligible-pay-stealth-credit-cards
+  [state side eid card]
+  (filter #(has-subtype? % "Stealth") (eligible-pay-credit-cards state side eid card)))
+
+(defn- total-available-stealth-credits
+  [state side eid card]
+  (->> (eligible-pay-stealth-credit-cards state side eid card)
+          (map #(+ (get-counters % :recurring)
+                   (get-counters % :credit)
+                   (-> (card-def %) :interactions :pay-credits ((fn [x] (:custom-amount x 0))))))
+          (reduce +)))
+
 ;; Credit
 (defmethod cost-name :credit [_] :credit)
 (defmethod value :credit [[_ cost-value]] cost-value)
+(defmethod stealth-value :credit [[_ __ stealth-amount]] (if (nil? stealth-amount) 0 stealth-amount))
 (defmethod label :credit [cost] (str (value cost) " [Credits]"))
 (defmethod payable? :credit
   [cost state side eid card]
-  (or (<= 0 (- (get-in @state [side :credit]) (value cost)))
-      (<= 0 (- (total-available-credits state side eid card) (value cost)))))
+  (and (<= 0 (- (total-available-stealth-credits state side eid card) (stealth-value cost)))
+       (<= 0 (- (value cost) (stealth-value cost)))
```

Comment:
Is this something we should be checking here or in the `min-stealth` function or at all? When would `stealth-value` be greater than the `value` of the cost?